gdal_subsetter/transform.py

```diff
@@ -353,57 +317,14 @@ def update_layernames(self, file_name, layer_names):
 
         dataset = None
 
-    def prepare_output_dir(self, output_dir):
-        """
-        Deletes (if present) and recreates the given output_dir, ensuring it exists
-        and is empty
-
-        Parameters
-        ----------
-        output_dir : string
-            the directory to delete and recreate
-        """
-        self.cmd2('rm', '-rf', output_dir)
-        self.cmd2('mkdir', '-p', output_dir)
-
-    def cmd2(self, *args) -> str:
-        """
-        This is used to execute the OS commands, such as cp, mv, and rm.
-        """
-        self.logger.info(
-            args[0] + ' ' + ' '.join(["'{}'".format(arg) for arg in args[1:]])
-        )
-        # result_str = check_output(args).decode('utf-8')
-        command = ' '.join(args)
-        exit_code=os.system(command)
-        if exit_code == 0:
-            exit_state = 'successfully'
-        else:
-            exit_state = 'failed'
-
-        return f'{command} executes {exit_state}.'
-
-    def cmd(self, *args) -> List[str]:
-        """ This is used to execuate gdal* commands. """
+    def execute_gdal_command(self, *args) -> List[str]:
```

Comment:
I'm still not a big fan of executing GDAL via building command line invocations and using `subprocess` to execute them. In future, we can implement some of the Python bindings from the `osgeo` package. For now, though, there's already enough in this PR.